cache/sizedlru/sizedlru.go

```diff
@@ -0,0 +1,111 @@
+// Package sizedlru implements an LRU cache that bounds the total space occupied
+// by the items it contains.
+
+package sizedlru
+
+import (
+	"container/list"
+)
+
+type SizedItem interface {
+	Size() int64
+}
+
+// SizedLRU is an LRU cache that will keep its total size below maxSize by evicting
+// items. If maxSize is zero, eviction is disabled.
+// SizedLRU is not thread-safe.
```

Comment:
Are you planning on wrapping every call with a mutex? I could see this becoming a bottleneck. Wdyt?

Reply:
Correct, but the critical section should only contain a few pointer swaps for the linked list and an hashtable lookup/update. This pales in comparison with all the work for setting up the HTTP connection and actually transfering the file. I'll be sure to test this with 3-4 gbit/s throughput.

Reply:
Yeah it's probably premature optimization. I am thinking we need benchmarks :-).

Reply:
Each request would:
1. lock the mutex
2. get the element from the lru, updating the eviction list
3. unlock the mutex
4. actually serve the file from disk

I would expect that the locking would be relatively unexpensive.